src/Serialization/Rings.jl

```diff
@@ -88,161 +95,185 @@ end
 
 ################################################################################
 # Multivariate and Universal Polynomials
-@registerSerializationType(QQMPolyRingElem)
-@registerSerializationType(ZZMPolyRingElem)
-@registerSerializationType(fqPolyRepMPolyRingElem)
-@registerSerializationType(zzModMPolyRingElem)
 @registerSerializationType(MPolyRingElem)
 @registerSerializationType(UniversalPolyRingElem)
 
 function save_internal(s::SerializerState, p::Union{UniversalPolyRingElem, MPolyRingElem})
     parent_ring = parent(p)
-    parent_ring = save_type_dispatch(s, parent_ring)
-    terms = []
+    base = base_ring(parent_ring)
+    parents = []
+    encoded_terms = []
 
     for i in 1:length(p)
-        term = Dict(
-            :exponent => save_type_dispatch(s, exponent_vector(p, i)),
-            :coeff => save_type_dispatch(s, coeff(p, i))
-        )
-        push!(terms, term)
+        if coeff == parent_ring(0)
```

Comment:
```suggestion
        if iszero(coeff(p, i))
```
Does this ever happen?

Reply:
this never happens and was removed